### DISCUSSION TITLE
Improve error handling of `klog edit`

### DISCUSSION THREAD

**Discussion by tscpp** [jotaen/klog#366] [Bugs]
I'm running a Windows 11 machine and I'm trying to get klog to open my files in vscode, but the files will just open in notepad. Klog doesn't "fail" to open the file, just not in the correct editor, and do not produce any output for me to debug.

- I have verified the paths is correct.
- I have tried to make the environment variable a string.
- I have tried using forward slashes.
- I have tried using a symlink pointing to code.exe.

```powershell
> $env:EDITOR="C:\Program Files\Microsoft VS Code\Code.exe"
> $env:EDITOR
C:\Program Files\Microsoft VS Code\Code.exe
> klog edit "C:\Users\elias\Desktop\time_records\xxx.klg"
```

**Comment by tscpp** [comment]
I tried to set $EDITOR to only "code" and it worked! I will keep this issue open, but feel free to close it.

**Comment by jotaen** [comment]
tldr: I think it works as expected right now, but the error handling should be improved here. So I’ll keep the issue open until that’s done.

When you run `klog edit`, it tries to open the file in various editors: the first try would be `$EDITOR`, and if that doesn’t succeed it falls back to default editors of the operating system, such as `notepad` for Windows, or `vim` for Linux. Only if none of that works, it would output an error.

What should happen instead is if the user specified `$EDITOR` and that fails, it should at least produce some sort of warning message instead of just skipping silently and disregarding the failure.

In regards to `code` vs. `C:\Program Files\Microsoft VS Code\Code.exe`: I’m not fully sure why the latter (full path) wouldn’t work, since I’m not super proficient with Windows. A potential problem might be the spaces in the path, though. But it’s good that the “short form” works out.

**Comment by jotaen** [comment]
A note for reference: since v5.4, the editor variable can now contain quoted words and spaces. E.g. on Powershell, you can now do:

```powershell
$env:EDITOR="'C:\Program Files\Sublime Text\subl.exe' --watch"
```

That would invoke the `C:\Program Files\Sublime Text\subl.exe` binary and pass `--watch` as argument to it.

You can optionally also use `$KLOG_EDITOR` now, if you want to set a different editor specifically for klog.
